source/distributed/tria_base.cc

```diff
@@ -191,11 +191,78 @@ namespace parallel
     }
 
     if (this->n_levels() > 0)
-      for (typename Triangulation<dim,spacedim>::active_cell_iterator
-           cell = this->begin_active();
-           cell != this->end(); ++cell)
-        if (cell->subdomain_id() == my_subdomain)
-          ++number_cache.n_locally_owned_active_cells[my_subdomain];
+      {
+        for (typename Triangulation<dim,spacedim>::active_cell_iterator
+             cell = this->begin_active();
+             cell != this->end(); ++cell)
+          if (cell->subdomain_id() == my_subdomain)
+            ++number_cache.n_locally_owned_active_cells[my_subdomain];
+
+        if (1)//settings & construct_multigrid_hierarchy)
```

Comment:
Obviously this needs to be changed. We only want to fill level_ghost_owners if we are running a MG computation, but tria_base.h doesn't have this setting. The only settings variable that both p::d::Tria and p::s::Tria share is the construct_multigrid_hierarchy. Is there a way that I could ask the triangulation here to tell me its settings? Or will I need to give the base class a member bool variable that stores whether or not this setting is turned on, and set it in p::d::T and p::s::T if the setting is turned on?

Reply:
That is annoying. No, sadly the settings are declared in ``p::d::Tria`` and ``p::s::Tria`` separately. Without doing large restructuring, I would do the following:
1. create a ``protected`` function ``fill_level_ghost_owners()`` inside the base class.
2. re-add the ``update_number_cache`` in the derived classes to do something like:
```
parallel::Triangulation<dim,spacedim>::update_number_cache();
if (settings & construct_multigrid_hierarchy)
  fill_level_ghost_owners();
```
I hope that works...